plugin.video.montreal.greek-tv/resources/lib/youtube.py

```diff
@@ -0,0 +1,352 @@
+                if '/default.jpg' in image:
+                    raise BaseException()
+                try:
+                    image = image.encode('utf-8')
+                except AttributeError:
+                    pass
+
+                self.list.append({'title': title, 'url': url, 'image': image})
+            except BaseException:
+                pass
+
+        return self.list
+
+    def _video_list(self, cid, url, pagination, limit):
+
+        try:
+            result = requests.get(url).text
+            result = json.loads(result)
+            items = result['items']
+        except BaseException:
+            pass
+
+        for i in list(range(1, limit)):
+
+            try:
+                if pagination is True:
+                    raise BaseException()
+                if not 'nextPageToken' in result:
+                    raise BaseException()
+                page = url + '&pageToken=' + result['nextPageToken']
+                result = requests.get(page).text
+                result = json.loads(result)
+                items += result['items']
+            except BaseException:
+                pass
+
+        try:
+            if pagination is False:
+                raise BaseException()
+            next = cid + '&pageToken=' + result['nextPageToken']
+        except BaseException:
+            next = ''
+
+        for item in items:
+            try:
+                title = item['snippet']['title']
+                try:
+                    title = title.encode('utf-8')
+                except AttributeError:
+                    pass
+
+                try:
+                    url = item['id']['videoId']
+                except KeyError:
+                    url = item['snippet']['resourceId']['videoId']
+
+                try:
+                    url = url.encode('utf-8')
+                except AttributeError:
+                    pass
+
+                image = item['snippet']['thumbnails']['high']['url']
+                if '/default.jpg' in image:
+                    raise BaseException()
+                try:
+                    image = image.encode('utf-8')
+                except AttributeError:
+                    pass
+
+                append = {'title': title, 'url': url, 'image': image}
+                if next != '':
+                    append['next'] = next
+                self.list.append(append)
+            except BaseException:
+                pass
+
+        try:
+            u = [list(range(0, len(self.list)))[i:i+50] for i in list(range(len(list(range(0, len(self.list))))))[::50]]
+            u = [','.join([self.list[x]['url'] for x in i]) for i in u]
+            u = [self.content_link % i + self.key_link for i in u]
+
+            threads = []
+            for i in list(range(0, len(u))):
+                threads.append(workers.Process(self.thread, u[i], i))
+                self.data.append('')
+            [i.start() for i in threads]
```

Comment:
launching a thread per item can exhaust the machine resources, you should limit the number of threads that can run at the same time by creating a thread pool. Queue tasks if they exceed the threadpool size.

Reply:
Removed.